### ISSUE TITLE
Provisioning issues with ansible on windows

### ISSUE THREAD

**Issue by mcheck** [evolution/genesis-wordpress#138]
Hey all,

Wondering if any of you have run into a similar error when provisioning a machine.
I'm using Windows 7, cygwin, virtualbox 4.3.14, vagrant 1.6.4 and genesis-wordpress 0.2.55
(Yes, I would rather be using a unix machine, but I don't have that luxury for this project :)

I get through booting the VM, but then it barfs claiming a permissions issue that I cannot reconcile:

```
<snip>
==> local: Updating /etc/hosts file on active guest machines...
==> local: Updating /etc/hosts file on host machine (password may be required)...
==> local: Running provisioner: hostmanager...
==> local: Running provisioner: shell...
    local: Running: inline script
==> local: stdin: is not a tty
==> local: dpkg-preconfigure: unable to re-open stdin: No such file or directory
==> local: Selecting previously unselected package software-properties-common.
==> local: (Reading database ...
==> local: 66829 files and directories currently installed.)
==> local: Unpacking software-properties-common (from .../software-properties-common_0.82.7.7_all.deb) ...
==> local: Setting up software-properties-common (0.82.7.7) ...
==> local: dpkg-preconfigure: unable to re-open stdin: No such file or directory
==> local: Selecting previously unselected package python-markupsafe.
==> local: (Reading database ...
==> local: 66843 files and directories currently installed.)
==> local: Unpacking python-markupsafe (from .../python-markupsafe_0.15-1_i386.deb) ...
==> local: Selecting previously unselected package python-jinja2.
==> local: Unpacking python-jinja2 (from .../python-jinja2_2.6-1ubuntu0.1_i386.deb) ...
==> local: Selecting previously unselected package sshpass.
==> local: Unpacking sshpass (from .../sshpass_1.05-1_i386.deb) ...
==> local: Selecting previously unselected package python-setuptools.
==> local: Unpacking python-setuptools (from .../python-setuptools_0.6.24-1ubuntu1_all.deb) ...
==> local: Selecting previously unselected package ansible.
==> local: Unpacking ansible (from .../ansible_1.7.1-precise-unstable-ppa_all.deb) ...
==> local: Processing triggers for man-db ...
==> local: Setting up python-markupsafe (0.15-1) ...
==> local: Setting up python-jinja2 (2.6-1ubuntu0.1) ...
==> local: Setting up sshpass (1.05-1) ...
==> local: Setting up python-setuptools (0.6.24-1ubuntu1) ...
==> local: Setting up ansible (1.7.1-precise-unstable-ppa) ...
==> local: Processing triggers for python-support ...
==> local: ERROR: The file localhost is marked as executable, but failed to execute correctly. If this is not supposed to be an executable script, correct this with `chmod -x localhost`.
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

I assume we're talking about `/provisioning/localhost`, but as it states, it is not executable outside the vm, but inside, it is.  I assume this has to do with using nfs as the sharing filesystem.

I tried forcing the permissions with: `mount_options: ["dmode=777", "fmode=666"]` but that messes up other files execution.

So, bottom line then, has anyone had experience running/provisioing genesis-wordpress on a windows  box that may have had similar issues?

Thanks!

**Comment by EvanK** [comment]
@AntJanus Do you recall if this `622` permission was necessary for *all* provisioning files, or just the inventory?

**Comment by TeodorMarinov** [comment]
pls fix this for Win users.

**Comment by EvanK** [comment]
There is a _totally untested_ fix for this (since I don't currently have access to a Windows machine), on the [138-provisiong-win-permhack](https://github.com/genesis/wordpress/compare/master...138-provisiong-win-permhack) branch.

Feedback is welcome

**Comment by EvanK** [comment]
Update: first commit on that branch actually broke on OSX/POSIX...so I doubt it fixed the issue in Windows. Second commit (b9faec7) has promise though

**Comment by TeodorMarinov** [comment]
Sorry but this patch does not work for me. The same error occurred. 
```
==> local: Processing triggers for python-support ...
==> local: ERROR: The file localhost is marked as executable, but failed to execute correct
y. If this is not supposed to be an executable script, correct this with `chmod -x localhos
`.
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

Skynet@T1000 /f/wp-genesis/src/test.dev (138-provisiong-win-permhack)</addr>
```
virtualbox 4.3.12, vagrant 1.6.5

**Comment by EvanK** [comment]
@TeodorMarinov Can you ssh into the vagrant machine and post what you get from `ls -al /vagrant/provisioning/`?

**Comment by TeodorMarinov** [comment]
```
vagrant@local:~$ ls -al /vagrant/provisioning/
total 11
drwxrwxrwx 1 vagrant vagrant 4096 Jun  1 20:57 .
drwxrwxrwx 1 vagrant vagrant 4096 Jun  1 20:57 ..
drwxrwxrwx 1 vagrant vagrant    0 Jun  1 20:57 files
drwxrwxrwx 1 vagrant vagrant    0 Jun  1 20:57 group_vars
-rwxrwxrwx 1 vagrant vagrant  101 Jun  1 20:57 local
-rwxrwxrwx 1 vagrant vagrant   49 Jun  1 20:57 localhost
-rwxrwxrwx 1 vagrant vagrant  106 Jun  1 20:57 production
-rwxrwxrwx 1 vagrant vagrant  922 Jun  1 20:57 provision.yml
-rwxrwxrwx 1 vagrant vagrant  103 Jun  1 20:57 staging
```

**Comment by EvanK** [comment]
One last try! In 88c69f9 we are actually _copying_ the inventory file to a wholly separate and NOT synced location, resetting its permissions, and then specifying _it_ as the inventory for ansible to use.

@TeodorMarinov Please try this out, and if it fails, let me know what output you get from `ls -al /tmp/ansible-inventory-localhost`

**Comment by AntJanus** [comment]
@EvanK I battled with that for a while. I wonder how this will turn out. At some point, I gave up and started dual-booting with Linux.

**Comment by EvanK** [comment]
@AntJanus Yeah, this is frustrating, even for a platform we don't "officially" support.

Evolution is using [native Ansible provisioning](http://docs.vagrantup.com/v2/provisioning/ansible.html), so hopefully won't suffer from this bug.  And I'm currently working on [a simpler upgrade path to Evolution](https://github.com/evolution/wordpress/issues/57), at which point we can retire Genesis once and for all.

**Comment by AntJanus** [comment]
Hey guys, I know this is a year too late but I figured out what to do. The `localhost` file is the issue but it's super-easy to remedy by altering it to be an executable Ruby script (host files *can* be executable!) like so:

```
#!/usr/bin/env ruby

require 'json'

module LT
  module Ansible
    def self.staging_inventory
      {
        webservers: {
          hosts: ["127.0.0.1"],
          vars: {
            ansible_connection: "local"
          }
        }
      }
    end
  end
end

if ARGV.find_index("--list") then
  puts LT::Ansible::staging_inventory.to_json
end
```
